pages/blog/integrating-mailchimp-with-your-nodejs-app.mdx

```diff
@@ -0,0 +1,350 @@
+
+For simplicity, we're logging in all the error messages into the browser console.
+
+Now, when we have set up our frontend, let's create our backend!
+
+It is time to make a `package.json` file to capture all the necessary modules required by our project in your repository.
+
+```
+$ npm init
+```
+
+This will ask you for a few project details and initiate a package.json file inside your repository.
+
+Now it is time to install all our necessary modules for the project. We will require the following modules:
+
+- `express`: Helps us define the routing table and set up middlewares to respond to HTTP Requests
+- `body-parser`: It is responsible for parsing the incoming request bodies in a middleware before we handle it.
+- `request`: Helps us make HTTP requests
+
+```
+$ npm install express body-parser request
+```
+
+Perfect, so now we can go ahead and get our hands dirty to write some actual code!
+
+Inside your project, create a new `app.js` file.
+
+Let’s do some basic setup here as well, getting our modules imported and connecting the middlewares.
+
+As a first step, let’s import all the modules we just installed:
+
+```
+const express = require("express");
+const bodyParser = require("body-parser");
+const request = require("request");
+```
+
+We’ll be needing the `path` as well.
+
+```
+const path = require("path");
+```
+
+Now, let’s define our application quickly.
+
+```
+const app = express();
+```
+
+Now, we need to define our middleware.
+
+1. To send the JSON data from the client, we will be using the `express.json()` middleware.
+
+```
+app.use(express.json());
+```
+
+2. To parse the data that is coming from the POST request, we will be using `body-parser`
+
+```
+app.use(bodyParser.urlencoded({extended: false}));
+```
+
+Note that we have set `extended` to `false`, which means that the values can only be `strings` or `arrays` which is what we need for now.
+
+3. Next up, we need to specify the folder which contains your frontend application and connect it to the express server. Here, we’re having a folder named `public`, which includes all our frontend code.
+
+```
+app.use(express.static(path.join(__dirname, 'public')))
+```
+
+4. Perfect, let's create our `POST` request quickly. We're just setting it up to add a `console.log()` to check if everything's working as expected.
+
+```
+app.post('/subscribe', (req, res) => {
+    console.log(“I am listening”);
+})
+```
+
+5. Now let's define the `PORT` which will host our application.
+
+```
+const PORT = process.env.PORT || 8888;
+```
+
+This sets the value to the environment `PORT` value in case we're using a cloud hosting or default it to 8888, which is our local machine endpoint.
+
+6. Finally, for all this to work, let's start a socket that listens for our connections on the `PORT`.
+
+```
+app.listen(PORT, console.log('The server has started!'));
+```
```

Comment:
This is tough to read in this way. I get that you're explaining one line at a time, but it's easy to get lost when I don't have the context of the full picture.

What I like to do in these scenarios is share a chunk of code, along with comments. And if anything requires more commentary than a single line, it's probably worth calling out _after_ sharing the snippet.

Reply:
I tried that but it wasn't looking very easy for a beginner to understand. To make it much clearer I can write the code towards the end. What say?

Reply:
It's not _where_ you're presenting it that makes it challenging, it's _how_ you're doing it. I understand that a large file can look intimidating, but I don't think that's what we're talking about here.

I think the information you've presented here is great. It's just missing context of where it fits in the larger file. Without some reference to how it all looks together, it's easy to get lost.

Reply:
I'll look into it and add something easier to read in that case